.github/workflows/code-climate-coverage-aggregation.yml

```
@@ -0,0 +1,85 @@
+name: Generate/Upload Spring Boot Coverage Report To Code Climate 
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+    
+jobs:
+  spring-boot:
+    env:
+      CC_TEST_REPORTER_ID: 8dfb70ce91b02659f7213078a8af0793b487eb85bc0f7854c6a9a4444058bb02
+    
+    name: Create, format and upload Java coverage artifact
+    runs-on: ubuntu-latest
+    
+    steps:
+      
+      # Checkout Project
+      - name: Checkout File Submission Repository
+        uses: actions/checkout@v2
+      
+      # Get Code Climate binary
+      - name: Download Code Climate Binary
+        run: curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
+      
+      # Permissions applied to the Code Climate Executable
+      - name: Apply executable perms to Code Climate Binary
+        run: chmod +x ./cc-test-reporter
+        
+      # Before build
+      - name: Before build
+        run: ./cc-test-reporter before-build
+
+      # Set required Git env vars for either pull request
+      - name: Set ENV for codeclimate (pull_request)
+        run: |
+          echo "GIT_BRANCH=${{ github.event.pull_request.head.ref }}" >> $GITHUB_ENV
+          echo "GIT_COMMIT_SHA=${{ github.event.pull_request.head.sha }}" >> $GITHUB_ENV
+        if: github.event_name == 'pull_request'
+
+      # Set required Git env vars for a push to master
+      - name: Set ENV for codeclimate (push)
+        run: |
+          echo "GIT_BRANCH=$GITHUB_REF" >> $GITHUB_ENV
+          echo "GIT_COMMIT_SHA=$GITHUB_SHA" >> $GITHUB_ENV
+        if: github.event_name == 'push'
+
+      # Trimming the ref to main in order to publish correct report (paambaati)
+      - name: Set ref/head/main to main
+        run: |
+          echo "GIT_BRANCH=master" >> $GITHUB_ENV
+        if: env.GIT_BRANCH == 'refs/heads/main'
+
+      # Setup Java Environment
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+        
+      # Where am I?
+      - name: Where am I before Maven?
+        run: |
+          pwd
+          echo "WHERE AM I?"
+          ls
+        
+      # Run Maven Verify to generate all jacoco reports
+      - name: Build with Maven
+        run: mvn -B verify
+      
+      # Formatting the BACKEND coverage reports generated (dynamically)
+      - name: Format BACKEND coverage reports
+        run: |
+          JACOCO_SOURCE_PATH=${{ github.workspace }}/mailservice-api/src/main/java ./cc-test-reporter format-coverage ${{github.workspace}}/target/site/jacoco/jacoco.xml --input-type jacoco --output coverage/efax-codeclimate.json;
```

```suggestion
          JACOCO_SOURCE_PATH=${{ github.workspace }}/src/main/java ./cc-test-reporter format-coverage ${{github.workspace}}/target/site/jacoco/jacoco.xml --input-type jacoco --output coverage/efax-codeclimate.json;
```
The project structure has been simplified - there is no longer a mailservice-api module, only the main project.

---

awesome I'll give this a shot:)